pdns/dynlistener.cc

```diff
@@ -283,7 +283,7 @@ string DynListener::getLine()
     mesg[len]=0;
   }
 
-  return &mesg[0];
+  return mesg.data();
```

Comment:
I would suggest applying this also, since adding a declaration in the if forest introduce the "whopsy" if-one-statement thing.
```diff
diff --git a/pdns/dynlistener.cc b/pdns/dynlistener.cc
index b408a5be4..352caf396 100644
--- a/pdns/dynlistener.cc
+++ b/pdns/dynlistener.cc
@@ -267,18 +267,24 @@ string DynListener::getLine()
     }
   }
   else {
-    if(isatty(0) != 0)
-      if(write(1, "% ", 2) !=2)
+    if (isatty(0) != 0) {
+      if (write(1, "% ", 2) != 2) {
         throw PDNSException("Writing to console: "+stringerror());
+      }
+    }
 
     ssize_t len = read(0, mesg.data(), mesg.size());
-    if (len < 0)
+    if (len < 0) {
       throw PDNSException("Reading from the control pipe: "+stringerror());
-    else if(len==0)
+    }
+
+    if (len == 0) {
       throw PDNSException("Guardian exited - going down as well");
+    }
 
-    if(static_cast<size_t>(len) == mesg.size())
+    if (static_cast<size_t>(len) == mesg.size()) {
       throw PDNSException("Line on control console was too long");
+    }
 
     mesg[len]=0;
   }
```

Reply:
That would make complete sense to me!

Reply:
Wow got jet-lagged and forgot your suggestion, will take into account later today. (github notification are silly...)